src/io/state-file/index.ts

```diff
@@ -100,18 +101,19 @@ async function restore(state: FileEntry[]): Promise<LoadResult[]> {
 
   const statuses: LoadResult[] = [];
 
+  const deserializeDataset = deserializeDatasetFiles(manifest, state);
   // We load them sequentially to preserve the order
-  for (const dataSet of dataSets) {
-    const files = state
-      .filter((entry) => entry.path === dataSet.path)
-      .map((entry) => entry.file);
+  // eslint-disable-next-line no-restricted-syntax
+  for (const dataset of datasets) {
+    // eslint-disable-next-line no-await-in-loop
+    const files = await deserializeDataset(dataset);
 
+    // eslint-disable-next-line no-await-in-loop
     const status = await datasetStore
-      .deserialize(dataSet, files)
+      .deserialize(dataset, files)
       .then((result) => {
         if (result.loaded) {
-          stateIDToStoreID[dataSet.id] = result.dataID;
-          fileStore.add(result.dataID, files);
```

Comment:
Do we not need to add to the file store anymore?

Reply:
`dicomStore` was adding files to `fileStore` in `importFiles`, `imageStore` was not (relying on wrapping code).  Now `imageStore` adds processed images to `fileStore`. So normalized to have final "datastores" be responsible for adding to `fileStore`